### PR TITLE
Add table oci_functions_function

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
**Integration test is not feasible**

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  display_name,
  id,
  application_id,
  lifecycle_state,
  image,
  image_digest
from
  oci_functions_function;
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+-----------------+------------------------------------------------+-------------------------------------------------------------------------+
| display_name | id                                                                                | application_id                                                                   | lifecycle_state | image                                          | image_digest                                                            |
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+-----------------+------------------------------------------------+-------------------------------------------------------------------------+
| hello-java   | ocid1.fnfunc.oc1.iad.aaaaaaaa7omirvbispjimid5xfo65cdoxiodddo5m7ug5xrsr7zafxln7ntq | ocid1.fnapp.oc1.iad.aaaaaaaar4i5qeq6cvv4uwvf2rhs2cn2kpihn3mpomr5zxiibpg2ruv5jwhq | ACTIVE          | iad.ocir.io/bmqeqvslavsz/test/hello-java:0.0.5 | sha256:ff4b362a19054efe35cdeade33b7e1085960eb655156830ccd2b91c663df0046 |
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+-----------------+------------------------------------------------+-------------------------------------------------------------------------+
> select
  display_name,
  id,
  application_id,
  trace_config -> 'isEnabled' as trace_config_is_enabled
from
  oci_functions_function
where
  not (trace_config -> 'isEnabled') :: bool;
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+-------------------------+
| display_name | id                                                                                | application_id                                                                   | trace_config_is_enabled |
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+-------------------------+
| hello-java   | ocid1.fnfunc.oc1.iad.aaaaaaaa7omirvbispjimid5xfo65cdoxiodddo5m7ug5xrsr7zafxln7ntq | ocid1.fnapp.oc1.iad.aaaaaaaar4i5qeq6cvv4uwvf2rhs2cn2kpihn3mpomr5zxiibpg2ruv5jwhq | false                   |
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+-------------------------+
> select
  display_name,
  id,
  application_id,
  memory_in_mbs
from
  oci_functions_function
where
  memory_in_mbs > 100;
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+---------------+
| display_name | id                                                                                | application_id                                                                   | memory_in_mbs |
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+---------------+
| hello-java   | ocid1.fnfunc.oc1.iad.aaaaaaaa7omirvbispjimid5xfo65cdoxiodddo5m7ug5xrsr7zafxln7ntq | ocid1.fnapp.oc1.iad.aaaaaaaar4i5qeq6cvv4uwvf2rhs2cn2kpihn3mpomr5zxiibpg2ruv5jwhq | 128           |
+--------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------+---------------+
```
</details>

### CODE REVIEW

File: oci/table_oci_functions_function.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+					Name:    "lifecycle_state",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "display_name",
+				Description: "The display name of the application.",
```
Comment (ParthaI) [inline oci/table_oci_functions_function.go:43]:
```suggestion
				Description: "The display name of the function.",
```

File: oci/table_oci_functions_function.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the function.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "application_id",
+				Description: "The OCID of the application the function belongs to.",
```
Comment (ParthaI) [inline oci/table_oci_functions_function.go:54]:
```suggestion
				Description: "The OCID of the function.",
```

File: oci/table_oci_functions_function.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+	// Build request parameters
+	request := buildFunctionsFilters(equalQuals)
+	request.ApplicationId = &applicationId
+	request.Limit = types.Int(50)
+	request.RequestMetadata = common.RequestMetadata{
+		RetryPolicy: getDefaultRetryPolicy(),
+	}
+
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
```
Comment (ParthaI) [inline oci/table_oci_functions_function.go:202]: Can we set the minimum limit as per api documentation and update the logic bit for handle the minimum limit.

File: oci/table_oci_functions_function.go [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+				Name:        "id",
+				Description: "The OCID of the function.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "application_id",
+				Description: "The OCID of the application the function belongs to.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ApplicationId"),
```
Comment (ParthaI) [inline oci/table_oci_functions_function.go:56]: Do we need this transform field?

Comment (c0d3r-arnab) [reply oci/table_oci_functions_function.go:56]: The data is not being displayed in the table otherwise. So I added it.

**Review by ParthaI — APPROVED** [review]
LGTM

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**